src/toil/batchSystems/slurm.py

```diff
@@ -447,11 +500,15 @@ def prepareSbatch(self,
             if gpus:
                 # This block will add a gpu supported partition only if no partition is supplied by the user
                 sbatch_line = sbatch_line[:1] + [f'--gres=gpu:{gpus}'] + sbatch_line[1:]
-                lowest_priority_gpus = sorted([i for i in self.boss.partitions if i["gres"]], key=lambda x: x["priority"])
-                new_partition_name = lowest_priority_gpus[0][0]  # get name of lowest priority partition that supports gpus
                 if not any(option.startswith("--partition") for option in sbatch_line):
                     # no partition specified, so specify one
-                    sbatch_line.append(f"--partition={new_partition_name}")
+                    # try to get the name of the lowest priority gpu supported partition
+                    lowest_gpu_partition = self.boss.partitions.default_gpu_partition
+                    if lowest_gpu_partition is None:
+                        # no gpu partitions are available, raise an error
+                        required_gpus = Requirer(dict(accelerator=parse_accelerator(gpus)))
+                        raise InsufficientSystemResources(required_gpus, "accelerators", [], details=["Could not find a Slurm partition that supports GPUs!"])
```

Comment:
I don't like that we're having to forge this since we don't actually have the original job anymore here. `InsufficientSystemResources` stringifies the Requirer input and uses that as a job name, so this will give us something like `The job <Requirer instance at wherever> is requesting 1 accelerators...` for our error message. Maybe we can't actually use `InsufficientSystemResources` here and should use a more generic error type.

```suggestion
                       raise RuntimeError(f"The job {jobName} is requesting GPUs, but the Slurm cluster does not appear to have an accessible partition with GPUs")
```